pkg/agent/controller/agent.go

```diff
@@ -288,10 +375,67 @@ func (a *Controller) newServiceImport(svcExport *lighthousev2a1.ServiceExport) *
 				originName:      svcExport.Name,
 				originNamespace: svcExport.Namespace,
 			},
+			Labels: map[string]string{
+				labelSourceName:      svcExport.Name,
+				labelSourceNamespace: svcExport.Namespace,
+				labelSourceCluster:   a.clusterID,
+			},
 		},
 	}
 }
 
+func (a *Controller) getServiceImport(svcExport *lighthousev2a1.ServiceExport) (*lighthousev2a1.ServiceImport, error) {
+	labelMap := map[string]string{
+		labelSourceName:      svcExport.Name,
+		labelSourceNamespace: svcExport.Namespace,
+		labelSourceCluster:   a.clusterID,
+	}
+	siListOptions := metav1.ListOptions{
+		LabelSelector: labels.Set(labelMap).String(),
+	}
+	serviceImports, err := a.lighthouseClient.LighthouseV2alpha1().ServiceImports(corev1.NamespaceAll).List(siListOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(serviceImports.Items) > 1 {
+		return nil, fmt.Errorf("Found multiple ServiceImports with local cluster ID.")
+	}
+
+	return &serviceImports.Items[0], nil
+}
+
+func (a *Controller) getIpsForService(service *corev1.Service, siType lighthousev2a1.ServiceImportType) []string {
+	if siType == lighthousev2a1.SuperclusterIP {
+		mcsIp := getGlobalIpFromService(service)
+		if mcsIp == "" {
+			mcsIp = service.Spec.ClusterIP
+		}
+
+		return []string{mcsIp}
+	}
+
+	endpoint, err := a.kubeClientSet.CoreV1().Endpoints(service.Namespace).Get(service.Name, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Endpoints for svc  (%s/%s) not found: %v", service.Namespace, service.Name, err)
```

Comment:
> So if it's not found, I assume we rely on the `endpointToRemoteServiceImport` to handle it. In this case we shouldn't log the error. Otherwise, I think we should return the error or nil or something to indicate to the caller to re-queue/retry.

Will change to debug, this was for my troubleshooting.

Reply:
bump

Reply:
Done